### PR TITLE
Delete Configuration Functionality

### LINKED ISSUE #268: Delete analysis configurations from the config sidebar
* Add a minus button to remove configurations
* Implement an `initial position` param on the AnalysisConfig object to secure config sidebar ordering
* Further clean up loading configs in order

### PR DESCRIPTION
**Pull request by ibrodycalixto**
Resolves #268, adds initialPosition attribute to AnalysisConfiguration object to standardize default configuration naming and loading order.

### CODE REVIEW

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js]: Since the initialPosition is not used when loading or saving the analysisMap, the highestPosition when you reload is just the initialPosition of whatever config was last edited - I tested this and was able to get configs with the same name after deleting and re-adding. I would either find a way to implement the loading order in this merge request (sounds hard) or just check when you load in which initialPosition is the highest (much easier).

File: leaf-ui/css/analysis_style.css
Comment (leshghi) [inline leaf-ui/css/analysis_style.css:361]: If you want you could use "cursor:pointer" here so it turns into the click-y hand that indicates it is a button

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js]: Make sure to update which config is highlighted as well to show/reflect which is the currAnalysisConfig in the UI

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js:549]: html whiz amazing work

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js]: The configElement here actually is the configDiv so you can delete this line

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js]: Instead of iterating through this list it might be easier to just use the jquery prev() and next() functions to get the id of the new currAnalysisConfig. You would just need to basically try "prev()" on the configElement first, and then if that returned null use "next()" instead. This would condense lines 373-387 into just two if statements, one nested inside the other which I think would be more concise and easier to follow. [Here is some documentation](https://www.w3schools.com/jquery/traversing_prev.asp) for prev(), next() is pretty much the same. You would just need to use 'div' as the selector too so it would be pretty simple.

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js]: Make sure all your comments have the space between the double slashes and first word and the first word is capitalized

**Review by leshghi — CHANGES_REQUESTED** [review]
This looks amazing thank you so much for taking on delete which is always the worst thing to implement! I have one comment on a bug that needs to be fixed before I officially approve, and the rest are just suggestions for condensing code/making it more efficient and readable. Thank you again sm this is great

File: leaf-ui/js/displayAnalysis.js
Comment (ibrodycalixto) [reply leaf-ui/js/displayAnalysis.js]: I made a fix to prevent duplicate names but would love to replicate this bug so I can see how it happened/how to fix.

File: leaf-ui/js/displayAnalysis.js
Comment (esterzhao) [inline leaf-ui/js/displayAnalysis.js]: Potentially allow user to delete all configs, but then repopulate with default empty Configuration1? ask @amgrubb for input

File: leaf-ui/js/displayAnalysis.js
Comment (esterzhao) [inline leaf-ui/js/displayAnalysis.js]: you should probably use that highestPosition var here just to be consistent

Comment (esterzhao) [reply leaf-ui/js/displayAnalysis.js]: also if you implement the "readd a default config when all are deleted", you could use this func and just add a line that resets highestPosition to 1 in this func

File: leaf-ui/js/displayAnalysis.js
Comment (esterzhao) [inline leaf-ui/js/displayAnalysis.js:288]: thank u for truly stellar commenting

File: leaf-ui/js/displayAnalysis.js
Comment (esterzhao) [inline leaf-ui/js/displayAnalysis.js]: remove this console log before merging

**Review by esterzhao — APPROVED** [review]
A couple nitpicky comments, and I added a few found bugs to the project, but this is good to merge once you pull changes from model-management!

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js]: This fix can actually go directly in the load analysis function since the bug is actually loading specific! I will explain more how to replicate in other comment thread

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js]: I think you can delete this TODO now since youve done it!

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js]: can delete this console.log

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [inline leaf-ui/js/displayAnalysis.js]: I think you can also delete this console.log unless you have a reason to leave it in!

**Review by leshghi — APPROVED** [review]
Amazing! One minor outstanding thing that can either be fixed here or pushed to the future: If you are on one config/have one config selected and then you go and delete a different config it will still switch you to the config directly under/over the deleted config. Would be cool to be able to delete a config without having the switch from the one you are currently on. Super minor and more a UI design choice/feature than a bug though so super low priority.

File: leaf-ui/js/displayAnalysis.js
Comment (leshghi) [reply leaf-ui/js/displayAnalysis.js]: Yes for sure! Its actually cause by saving and loading. So to replicate you would need to create some configs, rename any config that is not the last config, and then save. When you reload, since it saves/loads in order of most recently edited, it will load the last edited config (aka the config you renamed) last. Before you made the check this meant that the highest position would actually just be the initial position of the last loaded config, not necessarily the highest initial position. For example if you had config1, config2, config3, and then edited config2 to be named something else (lets say prev2), it would reload in the order config1, config3, prev2. So highest position would be saved as 2 from prev2, and the next config added would be named config3 and be a duplicate. Sorry I know that is a lot of words so let me know if it doesn't make sense!

**Comment by ibrodycalixto** [comment]
> Amazing! One minor outstanding thing that can either be fixed here or pushed to the future: If you are on one config/have one config selected and then you go and delete a different config it will still switch you to the config directly under/over the deleted config. Would be cool to be able to delete a config without having the switch from the one you are currently on. Super minor and more a UI design choice/feature than a bug though so super low priority.

I went ahead and implemented it!

**Pull request merged**
